### DISCUSSION TITLE
Duplicate access of settings file from multiple accounts

### DISCUSSION THREAD

**Discussion by samschultzponsys** [ThioJoe/ThioJoe-AHK-Scripts#23] [Q&A]
I know you mentioned config paths and such in your video, if I am using the exe and have set it up, what is the best way to duplicate the settings. Can I just drop a config file in a specific path? I would mostly use this as admin but sometimes under a limited user so I know that plays into install directories and config locations. Thanks again having that favorites bar is freaking awesome!

**Comment by ThioJoe** [comment]
Edit: Gonna convert this to a discussion

It will first look for the config file in the same directory as the script/exe. If it doesn't find it, it will then look for it in the user's AppData roaming directory in a folder called "Explorer-Dialog-Path-Selector".

Though you could change the AppData folder name or even the config file name in the script by editing these values):
```
global g_pth_SettingsFile := SettingsFile(
    "ExplorerDialogPathSelector-Settings.ini", ; Name of the settings file
    "Explorer-Dialog-Path-Selector"            ; Name of the folder in AppData where the settings file will be stored
)
```

If a settings file is loaded and you change and save the settings, it will update the file wherever it was loaded. If you save the settings when no settings file was loaded, then it will first try to save next to the script/exe, and then if it can't it will go in the AppData folder.

Assuming the following:
- The admin and user accounts have different usernames
- You want to share the same settings file, I can think of a few possibilities
- The exe is in Program Files

Then I can think of a few options:
- You could put the config file next to the Exe in program files. It will be able to read from it even when not escalated, but you'll have to escalate to update it. Whether that means while you're running the script as admin or copying an updated version of the config file and accepting the UAC prompt.
- Have two separate copies of the config file in each user's AppData folder
- **Probably the easiest:** Put the config in one user's AppData folder and make a Symlink for it in the other's AppData folder. So whether you run it from either account it will actually act on the same file.

**Reply by samschultzponsys** [reply]
Got it, in my scenario it would be in program files and called by a batch file on login most likely. I could use a shortcut to fire it as admin with runastool under a limited user if that wipes out the complexity.
